engine/src/lnxspec.cpp

```diff
@@ -670,37 +670,36 @@ char *MCS_resolvepath(const char *path)
 		}
 		delete tpath;
 	}
-	else
-		tildepath = strclone(path);
+    else if (path[0] != '/')
+    {
+        // SN-2015-06-05: [[ Bug 15432 ]] Fix resolvepath on Linux: we want an
+        //  absolute path.
+        char *t_curfolder;
+        t_curfolder = MCS_getcurdir();
+        tildepath = new char[strlen(t_curfolder) + strlen(path) + 2];
+        /* UNCHECKED */ sprintf(tildepath, "%s/%s", t_curfolder, path);
+
+        delete t_curfolder;
+    }
+    else
+        tildepath = strclone(path);
 
 	struct stat64 buf;
 	if (lstat64(tildepath, &buf) != 0 || !S_ISLNK(buf.st_mode))
 		return tildepath;
-	int4 size;
-	char *newname = new char[PATH_MAX + 2];
-	if ((size = readlink(tildepath, newname, PATH_MAX)) < 0)
-	{
-		delete tildepath;
-		delete newname;
-		return NULL;
-	}
-	delete tildepath;
-	newname[size] = '\0';
-	if (newname[0] != '/')
-	{
-		char *fullpath = new char[strlen(path) + strlen(newname) + 2];
-		strcpy(fullpath, path);
-		char *sptr = strrchr(fullpath, '/');
-		if (sptr == NULL)
-			sptr = fullpath;
-		else
-			sptr++;
-		strcpy(sptr, newname);
-		delete newname;
-		newname = MCS_resolvepath(fullpath);
-		delete fullpath;
-	}
-	return newname;
+
+    char *newname = new char[PATH_MAX + 2];
+
+    // SN-2015-06-05: [[ Bug 15432 ]] Use realpath to solve the symlink.
+    if (realpath(tildepath, newname) == NULL)
+    {
+        // Clear the memory in case of failure
+        delete newname;
+        newname = NULL;
+    }
+
+    delete tildepath;
+    return newname;
```

Comment:
I chose to stick to returning NULL since it is also done if the path starts with `~`, but does not match either `~/<path>` or `~<user>/<path>` (line 663).
I'm not sure that returning an unresolved path, when asking to resolve the path, is really what we want to do (silent errors are better to be avoided, I would say), but that I leave you deciding on this :)